stetho-sample/src/main/java/com/facebook/stetho/sample/APODRssFetcher.java

```diff
@@ -73,23 +75,46 @@ public void onFailure(IOException e) {
       // Show in Stetho :)
     }
 
-    private void parseAndStore(byte[] body)
-        throws
-            XmlPullParserException,
-            RemoteException,
-            OperationApplicationException,
-            IOException {
+    private List<RssItem> parseRss(byte[] body) throws IOException, XmlPullParserException {
       XmlPullParser parser = Xml.newPullParser();
       parser.setInput(new ByteArrayInputStream(body), "UTF-8");
       List<RssItem> items = new RssParser(parser).parse();
       Log.d(TAG, "Fetched " + items.size() + " items");
 
+      return items;
+    }
+
+    public List<ApodItem> decorateRssItemsWithLinkImages(List<RssItem> rssItems) {
+      ArrayList<ApodItem> apodItems = new ArrayList<>(rssItems.size());
+      final CountDownLatch fetchLinkLatch = new CountDownLatch(rssItems.size());
+      for (RssItem rssItem : rssItems) {
+        final ApodItem apodItem = new ApodItem();
+        apodItem.rssItem = rssItem;
+        fetchLinkPage(rssItem.link, new PageScrapedCallback() {
+          @Override
+          public void onPageScraped(@Nullable List<String> imageUrls) {
+            apodItem.largeImageUrl = imageUrls != null && !imageUrls.isEmpty()
```

Comment:
If it can't fetch the URL provided it uses `null` to indicate this instead of an empty list which communicates that it fetched the page but it had no `<img>` tags in it.  The distinction is ignored by the APOD-specific code, but the lower level code reads more clearly IMO having a distinction.